lang/syn/src/parser/context.rs

```diff
@@ -40,6 +40,53 @@ impl CrateContext {
             modules: ParsedModule::parse_recursive(root.as_ref())?,
         })
     }
+
+    // Perform Anchor safety checks on the parsed create
+    pub fn safety_checks(&self) -> Result<(), anyhow::Error> {
+        // Check all structs for unsafe field types, i.e. AccountInfo and UncheckedAccount.
+        for (_, ctx) in self.modules.iter() {
+            for unsafe_field in ctx.unsafe_struct_fields() {
+                // Check if unsafe field type has been documented with a /// SAFETY: doc string.
+                let is_documented_or_safe = unsafe_field.attrs.iter().any(|attr| {
+                    attr.tokens.clone().into_iter().any(|token| match token {
+                        // Check for comments containing SAFETY: or CHECK:
+                        proc_macro2::TokenTree::Literal(s) => {
+                            s.to_string().contains("SAFETY:") || s.to_string().contains("CHECK:")
```

Comment:
We should probably choose one or the other. I don't have a preference.

Reply:
I suspect rustaceans will be confused with `SAFETY:`, since it normally refers to the use of `unsafe {...}`

Reply:
CHECK sounds good then 👍

I'll also remove the colon in the string check because I often find myself doing `/// CHECK something` in tests and then being confused about the error.